[DX-1942] Gateway Release Notes 5.8.1 (#6288)
TykTechnologies/tyk-docs: tyk-docs/content/developer-support/release-notes/gateway.md

---
## 5.8 Release Notes

### 5.8.1 Release Notes

> #### Release Date XXX

Reviewer: ```suggestion
#### Release Date 9 May 2025
```

#### Release Highlights

> This release focuses mainly on bug fixes. For a comprehensive list of changes, please refer to the detailed [changelog]({{< ref "#Changelog-v5.8.1" >}}) below.

Reviewer 2: No - please use the new wording. This does not focus "mainly" on bug fixes.

#### Breaking Changes

There are no breaking changes in this release.

#### Dependencies {#dependencies-5.8.1}

##### Compatibility Matrix For Tyk Components

| Gateway Version | Recommended Releases | Backwards Compatibility |
|----    |---- |---- |

> | 5.8.1 | MDCB v2.8.0     | MDCB v2.8.0 |

Reviewer 2: I believe we are releasing MDCB 2.8.1 with this version

| | EDP v1.13 | EDP all versions |
| | Pump v1.12.0 | Pump all versions |
| | TIB (if using standalone) v1.7.0 | TIB all versions |

##### 3rd Party Dependencies & Tools

| Third Party Dependency                                       | Tested Versions        | Compatible Versions    | Comments | 
| ------------------------------------------------------------ | ---------------------- | ---------------------- | -------- | 
| [Go](https://go.dev/dl/)                                     | 1.23  |  1.23  | [Go plugins]({{< ref "api-management/plugins/golang" >}}) must be built using Go 1.23 | 
| [Redis](https://redis.io/download/)  | 6.2.x, 7.x  | 6.2.x, 7.x  | Used by Tyk Gateway | 
| [OpenAPI Specification](https://spec.openapis.org/oas/v3.0.3)| v3.0.x                 | v3.0.x                 | Supported by [Tyk OAS]({{< ref "api-management/gateway-config-tyk-oas" >}}) |

Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this release.

> #### Upgrade instructions {#upgrade-5.8.0}

Reviewer 2: ```suggestion
#### Upgrade instructions {#upgrade-5.8.1}
```

If you are upgrading to 5.8.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.8.1)
  - ```bash
    docker pull tykio/tyk-gateway:v5.8.1
    ``` 
- Helm charts
  - [tyk-charts v3.0.0]({{<ref "developer-support/release-notes/helm-chart#300-release-notes" >}})

- [Source code tarball for OSS projects](https://github.com/TykTechnologies/tyk/releases)

#### Changelog {#Changelog-v5.8.1}

##### Fixed

<ul>
<li>
<details>
<summary>Fixed Inconsistent Context Behavior in UDG APIs</summary>

> Fixed a bug where headers being sent upstream in UDG APIs were not being cached accurately, ensuring correct and consistent context handling for all requests.

Reviewer 2: Please ensure consistent wording with the 5.3.11 release notes, which contained this fix.

</details>
</li>
<li>
<details>
<summary>Improved Route Matching Logic for API Requests</summary>

Resolved an issue where some API requests were routed incorrectly due to path sorting not prioritizing specific routes over parameterized ones.
</details>
</li>
<li>
<details>

> <summary>Resolved Incorrect Timeout Application for Specific API Endpoints</summary>

Reviewer: ```suggestion
<summary>Resolved Issue With Default Enforced Request Timeout</summary>
```

> Fixed an issue where proxy_default_timeout for specific API endpoints were not being applied correctly, causing requests to use the global timeout instead.

Reviewer: ```suggestion
Fixed an issue where an [enforced timeout]({{< ref "tyk-self-managed#enforced-timeouts" >}}) set for a specific API endpoint could be overruled by the configured [proxy_default_timeout]({{< ref "tyk-oss-gateway/configuration#proxy_default_timeout" >}}). Now if an endpoint-level timeout is set then this will be honoured, regardless of any default timeout that is configured.
```

</details>
</li>
<li>
<details>

> <summary>Fixed race condition in gateway license allocation</summary>

Reviewer: ```suggestion
<summary>Fixed Issue With Tyk Self-Managed Gateways Claiming Licenses</summary>
```

Resolved a race condition in gateway license allocation that occasionally allowed more or fewer gateways than licensed to register and serve traffic.
</details>
</li>
<li>
<details>
<summary>Resolved merging issue in field-based policy permissions</summary>

> Resolved a bug where allowed_types from multiple policies were incorrectly merged using intersection logic. Policies now correctly merge fields to allow access to any fields listed across the applied policies.

Reviewer: ```suggestion
Resolved a bug where `allowed_types` from multiple policies were incorrectly merged using intersection logic. Policies now correctly merge fields to allow access to any fields listed across the applied policies.
```
